directives/run-async/test/run-async.test.js

```diff
@@ -0,0 +1,112 @@
+import { expect, fixture } from '@open-wc/testing';
+import { InitialStateError, runAsync } from '../run-async.js';
+import { html } from 'lit-element/lit-element.js';
+
+describe('run-async directive', () => {
+
+	let taskKey, taskInfo, taskResolve, taskReject;
+	function task(key, info) {
+		taskKey = key;
+		taskInfo = info;
+		return new Promise((res, rej) => {
+			taskResolve = res;
+			taskReject = rej;
+		});
+	}
+
+	function createFixture(key = 'some-key', pendingState = true) {
+		return fixture(html`<p>${runAsync(key, task, {
+			initial: () => html`initial`,
+			failure: () => html`failure`,
+			pending: () => html`pending`,
+			success: () => html`success`
+		},
+		{ pendingState: pendingState })}</p>`);
+	}
+
+	afterEach(() => {
+		taskKey = undefined;
+		taskInfo = undefined;
+		taskResolve = undefined;
+		taskReject = undefined;
+	});
+
+	describe('task execution', () => {
+
+		it('should pass key and abort controller to task', async() => {
+			await createFixture('my-key');
+			expect(taskKey).to.equal('my-key');
+			expect(taskInfo.signal).to.not.be.undefined;
+		});
+
+	});
+
+	describe('templates', () => {
+
+		it('should render initial template if InitialStateError occurs', async() => {
+			const elem = await createFixture();
+			taskReject(new InitialStateError());
+			await elem.updateComplete;
+			expect(elem.innerText).to.equal('initial');
+		});
+
+		// TODO: was this actually desired behaviour, or should nothing be rendered?
```

Comment:
I wasn't totally sure if this was intended, but it reflects [how the original was written](https://github.com/PolymerLabs/async-demos/blob/master/packages/await-io/src/run-async.ts). Essentially if an `initial` template isn't provided and an `InitialStateError` happens it's going to render the `failure` template instead.

Reply:
That seems reasonable.

Reply:
Cool thanks, I've removed the comment. This PR is now fully ready for review.